wrapanapi/msazure.py

```diff
@@ -627,30 +654,47 @@ def is_stack_empty(self, stack_name):
     def info(self):
         pass
 
-    def remove_unused_blobs(self):
+    def remove_unused_blobs(self, resource_group=None):
         """
         Cleanup script to remove unused blobs: Managed vhds and unmanaged disks
+        Runs though all storage accounts in 'resource_group'
         Returns list of removed disks
         """
-        # removing unmanaged disks
-        removed_blobs = []
-        container_client = BlockBlobService(self.storage_account, self.storage_key)
-        for container in container_client.list_containers():
-            for blob in container_client.list_blobs(container_name=container.name, prefix='test'):
-                if blob.properties.lease.status == 'unlocked':
-                    self.logger.info("Removing Blob {b} "
-                                     "from containter {c}".format(b=blob.name, c=container.name))
-                    container_client.delete_blob(container_name=container.name,
-                                                 blob_name=blob.name)
-                    removed_blobs.append({'container': container.name, 'blob': blob.name})
+        removed_blobs = {}
+        self.logger.info('Attempting to List unused disks/blobs')
+        resource_group = resource_group or self.resource_group
+        removed_blobs[resource_group] = {}
+        for storage_account in self.list_storage_accounts_by_resource_group(resource_group):
+            self.logger.info('Checking storage account "{}":'.format(storage_account))
+            removed_blobs[resource_group][storage_account] = {}
+            # removing unmanaged disks
+            key = self.get_storage_account_key(storage_account, resource_group)
+            container_client = BlockBlobService(storage_account, key)
```

Comment:
I'll keep this code using it's own container_client as we need to loop though all containers and leave the default as is